### PR TITLE
Enable API Gateway v2 (HTTP) for expose

### PR DESCRIPTION
**Pull request by gordon-klotho**
This change enables HTTP API Gateways to be created with routes set up.

### Standard checks

- **Unit tests**: IaC is not unit tested
- **Docs**: 🚧 
- **Backwards compatibility**: ✅

### CODE REVIEW

File: pkg/infra/pulumi_aws/iac/api_gateway.ts [hunk 1]
```diff
@@ -40,105 +42,154 @@ export class ApiGateway {
...
+        switch (execUnit.type) {
+            case 'ecs':
+                const ecsNlb = this.lib.execUnitToNlb.get(r.execUnitName)!
+                return new aws.apigatewayv2.Integration(
+                    integrationName,
+                    {
+                        apiId: api.id,
+                        integrationType: 'HTTP_PROXY',
+                        integrationMethod: 'ANY',
+                        integrationUri: ecsNlb.loadBalancer.arn,
```
Comment (jhsinger-klotho) [inline pkg/infra/pulumi_aws/iac/api_gateway.ts:113]: this will only work if the nlb is an awsx nlb which i think we moved completely off of now

Comment (jhsinger-klotho) [reply pkg/infra/pulumi_aws/iac/api_gateway.ts:113]: nevermind

Comment (jhsinger-klotho) [reply pkg/infra/pulumi_aws/iac/api_gateway.ts:113]: i thought we migrated away from it but we didnt

**Review by jhsinger-klotho — APPROVED** [review]

**Review by DavidSeptimus-Klotho — APPROVED** [review]

**Comment by DavidSeptimus-Klotho** [comment]
Do we want to update the signature of `core.NewGateway()` to require an `ApiType` value?

**Comment by gordon-klotho** [comment]
> Do we want to update the signature of `core.NewGateway()` to require an `ApiType` value?

`core.Gateway` wasn't changed, only the config types so no update necessary

**Pull request merged**